src/or/circuitlist.c

```diff
@@ -1844,6 +1845,58 @@ circuit_mark_all_dirty_circs_as_unusable(void)
   SMARTLIST_FOREACH_END(circ);
 }
 
+/**
+ * Report any queued cells on or_circuits as written in our bandwidth
+ * totals, for the specified channel direction.
+ *
+ * When we close a circuit or clear its cell queues, we've read
+ * data and recorded those bytes in our read statistics, but we're
+ * not going to write it. This discrepancy can be used by an adversary
+ * to infer information from our public relay statistics and perform
+ * attacks such as guard discovery.
+
+ * See https://trac.torproject.org/projects/tor/ticket/23512.
+ */
+void
+circuit_synchronize_written_or_bandwidth(const circuit_t *c,
+                                         circuit_channel_direction_t dir)
+{
+  size_t cells;
+  size_t cell_size;
+  size_t written_sync;
+  const channel_t *chan = NULL;
+  const or_circuit_t *or_circ;
+
+  if (!CIRCUIT_IS_ORCIRC(c))
+    return;
+
+  or_circ = CONST_TO_OR_CIRCUIT(c);
+
+  if (dir == CIRCUIT_N_CHAN)
+    chan = c->n_chan;
+  else
+    chan = or_circ->p_chan;
+
+  if (dir == CIRCUIT_N_CHAN)
+    cells = c->n_chan_cells.n;
+  else
+    cells = or_circ->p_chan_cells.n;
```

Comment:
Instead of adding the `circuit_channel_direction_t` and doing these if/else, can't we simply pass the channel to the function?

Reply:
Nvm here... The queue is on the circuit, not the channel so we need the direction to know which queue to get.